### ISSUE TITLE
Magicpin - compliance check

### ISSUE THREAD

**Issue by BLR-0118** [ONDC-Official/v1.1.0-logs#53]
Flow 1
====
1. /on_search:
- bpp/providers.time.timestamp must match Context.timestamp;
- fulfillments.contact.phone can be 10 or 11 digits only (no "+");
- time.schedule.holidays is mandatory (can be empty);
2. /on_init, /on_confirm:
- billing.created_at, billing.updated_at timestamp should remain the same as what it was set in /init (since buyer app sets it). This is being changed in /on_init, /on_confirm, etc.;
3. /on_confirm:
- order.created_at should remain the same as what is set in /confirm (as buyer app creates it);
4. /on_status_delivered_flow:
- both pickup time (fulfillment.start.time.timestamp) and delivery time (fulfillment.end.time.timestamp) needs to be set;

Flow 2
====
1. /on_init_not_serviceable_flow:
- since this is not serviceable, there should be no delivery charge;

Flow 5
====
1. /on_cancel_flow: if order is cancelled, value (quote.price) should become 0;

**Comment by BLR-0118** [comment]
@Tarosh

**Comment by BLR-0118** [comment]
1. /on_search:
- contact_details_consumer_care should be in this format (as per contract): "name,email,contactno";
2. /on_init:
- billing.created_at & updated_at should remain the same as what was entered in /init (since buyer app owns this);
- payment block: only keep buyer finder fee type & amount, settlement_details;
3. /on_status_delivered_flow_1:
- pickup time (fulfillment.start.time.timestamp) is updated here and in this case, pickup time is > delivery time, which is incorrect;
- pickup time will be set when fulfillment state code becomes "Order-picked-up" and shouldn't change after that;
- order.updated_at which is provided in the different on_status APIs are not getting updated;
4. Why is /cancel API not supported?
- can Magicpin support /cancel for orders that are in "Created" state (i.e. not yet accepted by merchant)

**Comment by BLR-0118** [comment]
@tarosh3

**Comment by tarosh3** [comment]
@BLR-0118 Magicpin does not support cancellation once the order is accepted by the merchant.

**Comment by BLR-0118** [comment]
1. Pickup time (fulfillment.start.time.timestamp) in on_status_delivered_flow_1 is different from on_status_picked_flow_1. Why?
2. /on_confirm shows order.state = "Created", i.e. not auto-accepted by merchant. In this state only, won't you support order cancellation? For this, you need the cancel API as well.

**Comment by BLR-0118** [comment]
@tarosh3

**Comment by tarosh3** [comment]
1. The fulfillment.start.time.timestamp issue is now fixed. fulfillment.start.time.timestamp will be updated on picked up flow and remains the same at delivery status.

2.For some merchants the order is being auto accepted and for some the merchant have to accept the order manually. So we are sending an unsolicited on_status call with state as "Accepted"

We are marking all the items as non cancellable. So we don't support cancellation as of now. But we have completed the flow for cancel as well. So in future, if we have to make the items as cancellable we can do that too.
Also, I have submitted the logs for the cancel flow as well.

@BLR-0118
